actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -114,16 +117,25 @@ module ClassMethods
       #
       #   If you need to add verification to the beginning of the callback chain, use <tt>prepend: true</tt>.
       # * <tt>:with</tt> - Set the method to handle unverified request.
+      # * <tt>:token_store</tt> - Set the strategy of storing the CSRF token. You can supply a hash with custom options
+      #   (if the strategy supports that), such as a cookie name, an object, inheriting from
+      #   ActionController::RequestForgeryProtection::TokenStoreStrategies::TokenStore.
       #
       # Valid unverified request handling methods are:
       # * <tt>:exception</tt> - Raises ActionController::InvalidAuthenticityToken exception.
       # * <tt>:reset_session</tt> - Resets the session.
       # * <tt>:null_session</tt> - Provides an empty session during request but doesn't reset it completely. Used as default if <tt>:with</tt> option is not specified.
+      #
+      # You can choose from three token storage strategies or pass your own:
+      # * <tt>:session_store</tt> – the classic strategy, the token is stored within the session.
+      # * <tt>:cookie_store</tt> – stores the token in a cookie, separate from the session.
+      # * <tt>:fallback_store</tt> – a temporary store that allows you to smoothly switch between different stores.
       def protect_from_forgery(options = {})
         options = options.reverse_merge(prepend: false)
 
         self.forgery_protection_strategy = protection_method_class(options[:with] || :null_session)
         self.request_forgery_protection_token ||= :authenticity_token
+        self.token_store = build_token_store_strategy(options[:token_store] || :session_store)
```

Comment:
I think this should be parameterized in the same way the `forgery_protection_strategy` is, instead of a global config param. That will allow users to set different behaviors on a per-controller basis.